src/components/QueryEditor.tsx

```diff
@@ -126,7 +132,7 @@ export function QueryEditor({ query, onChange, onRunQuery, datasource }: Props)
               label: 'Submit query',
               keybindings: [monaco.KeyMod.CtrlCmd | monaco.KeyCode.Enter],
               run: async function (ed) {
-                onQueryTextChange(ed.getValue());
+                onChange({ ...query, apl: ed.getValue(), totals: totals.current });
```

Comment:
I'm a bit nervous because in the onQueryTextChange there's a function call and now it isn't made anymore

Reply:
You mean we should call `setQueryStr(ed.getValue())` in there?

Reply:
Fair, added in 24ecc88